src/main/java/org/opensearch/sdk/rest/DeprecatedRouteHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.rest;
+
+import java.util.function.Function;
+
+import org.opensearch.extensions.rest.ExtensionRestResponse;
+import org.opensearch.rest.RestHandler.DeprecatedRoute;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestRequest.Method;
+
+/**
+ * A subclass of {@link DeprecatedRoute} that includes a handler method for that route.
+ */
+public class DeprecatedRouteHandler extends DeprecatedRoute {
```

Comment:
Can we have this class as a nested class in `ExtensionRestHandler`?

Reply:
I'm on the fence about this one.  If we want users to use it they need to see it.  Whatever we do with this we should do with `RouteHandler`.